libs-scala/resources/src/main/scala/com/digitalasset/resources/AbstractResourceOwner.scala

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.resources
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+/**
+  * A ResourceOwner of type [[A]] is can acquire a [[Resource]] of the same type and its operations
+  * are applied to the [[Resource]] after it has been acquired.
+  *
+  * @tparam A The [[Resource]] value type.
+  */
+@FunctionalInterface
```

Comment:
Are you doing this to have some kind of interoperability with Java? With `acquire` using an implicit parameter I'm not entirely sure of the usefulness of this.

Reply:
Yeah, this probably isn't true anyway. I shall delete it. (It was there before.)